app/models/step.rb

```diff
@@ -12,18 +12,18 @@ class Approval < ActiveRecord::Base
   acts_as_list scope: :proposal
   validates :proposal, presence: true
 
-  belongs_to :parent, class_name: 'Approval'
-  has_many :child_approvals, class_name: 'Approval', foreign_key: 'parent_id', dependent: :destroy
-
-  scope :individual, -> { where(type: 'Approvals::Individual') }
+  belongs_to :parent, class_name: "Step"
+  has_many :child_approvals, class_name: "Step", foreign_key: "parent_id", dependent: :destroy
 
+  # @TODO: Auto-generate list of subclasses
+  scope :individual, -> { where(type: ["Steps::Approval", "Steps::Purchase"]).order("position ASC") }
```

Comment:
Interesting - how would it work as a separate scope? Would I be able to combine the two?